libs/datamodel/core/src/transform/ast_to_dml/db/relations.rs

```diff
@@ -0,0 +1,219 @@
+use super::{context::Context, types::RelationField};
+use crate::ast;
+use std::collections::BTreeSet;
+
+#[derive(PartialOrd, Ord, PartialEq, Eq, Debug, Clone)]
+enum OneToManyRelationFields {
+    Forward(ast::FieldId),
+    Back(ast::FieldId),
+    Both(ast::FieldId, ast::FieldId),
+}
```

Comment:
Any plans to do the same split for 1:1 relations?

Reply:
I should have written down my understanding of how we infer 1:1 relations because it could be wrong.

The logic I have in mind and what I implemented is to look at the relation fields with the same relation name on both sides.

- If only one is there, infer a 1:m relation.
- If both are there, it depends on the arity:
  - Both lists: implicit many-to-many relation
  - One list, one other: 1:m relation
  - Both required/optional: 1:1 relation.
 
Additionally, I think that in the case of 1:1 relations, there should be additional logic.

- Both required is forbidden
- One required, one optional: model A is the one with the required field (corresponding to the foreign key)
- Both optional: ???? (we should clarify this before we merge this PR — my guess would be the relation field with a `fields` or `references` argument, otherwise error)

Does it sound right to you?

Reply:
This sounds right for me. We should have enough tests to validate any breakage...

Reply:
I think you can have a 1:1 relation also if you only have one field. If it is the one holding the reference.

Reply:
Aah, if the underlying scalar fields are unique?